Welding Tool Updates

Nox38 opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
This PR changes a few things about welders:
- Engineering toolbelts now start with industrial welders instead of normal ones, giving them 2.5x as much fuel to work with.
- Lit welders are now ginormous items instead of large items, preventing you from putting a lit welder in your backpack.
- Emergency welder's fuel capacity has been increased from 50 to 65, giving it just enough to weld two walls in a row.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
The reasons for resizing the industrial, experimental and advanced welders is to distinguish them from the other welders and make their sprites actually fit inside of the grid inventory.
Industrial and advanced industrial welders are not buildable or purchasable in game, being exclusively mapped as gamerloot on certain maps. The limited fuel capacity of normal welders forces engineers to rely on RCDs or take a lot of breaks to refill during big repairs, and I feel it makes sense to give them something a cut above the standard welder.
Emergency welders have always been pretty awful, but the fact that they are about 5 fuel away from being able to weld two walls in a row feels more like a bug than a feature.

## Technical details
<!-- Summary of code changes for easier review. -->

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
:cl: Nox38
- tweak: Engineers now start with industrial welders. Emergency welding tools have slightly more fuel.

iaada commented:
A sticking point for engineers is that now there's no room on the belt for your RCD. I'm in favor of the change though

Lukasz825700516 commented:
Cool giving engineers better welders, but why we should not be able to put lit welders in the bag, if we can do that with energy swords?

ArtisticRoomba commented:
Is there a reason why this PR is a draft?

NoElkaTheGod commented:
> Cool giving engineers better welders, but why we should not be able to put lit welders in the bag, if we can do that with energy swords?

Why would you even want to do that? I'd say this should be removed from the esword as well. It makes no sense both mechanicaly (why store a lit welder if you can just turn it off) and realisticaly (you'll just burn your backpack by doing that).

Nox38 commented:
> Cool giving engineers better welders, but why we should not be able to put lit welders in the bag, if we can do that with energy swords?

Making lit energy swords bigger was outside the scope of this PR, im hoping to do that in a future one. If a maintainer says its fine ill change that here.

Nox38 commented:
> Engineers are already pretty strapped for space, and while I do agree that the welders do look a bit silly in terms of size, a 2x2 area being taken up reduces the effective amount of equipment an Engineer can carry, and I think this outweighs any benefit gained by giving engineers the advanced welder.
> They are arguably pretty distinguished already, welding faster and having more fuel, with the experimental welder having fuel regeneration.

Would reverting the sizes fix this? Or a corner 3 tile size like you had me do with the price gun?

> They can be found uncommonly in welding lockers and other lockers, so it's a good thing to find when maints diving.

That makes sense as an upgrade for tiders, but shouldn’t engineers have something slightly better simply for being.. engineers? A different way to do this could be a handheld or easy to drag portable fuel tank I guess.

> I mean, emergency welders are used in an emergency - they're not really supposed to be used long-term for anything else. Moreso welding doors, busting down one wall, or a window to get out of something. As such they're pretty bad.

Ive never seen someone use an emergency welder in a emergency due to the extremely long doafters and fact that it will still fuck up your eyes. I was toying with the idea of making them non refillable but safe to use without protection, what do you think about that?

> Also I ask you to please follow our PR guidelines on draft PRs.

Sorry, I was going to mark it open after the tests finished but went to bed.

Nox38 commented:
> I thought engies put them in their bag (gas analyzer and t-ray in belt instead)

T-ray goes in your pocket, and either the holofan & gas analyzer or RCD in the belt.

> Anyway I would just give engi belts a starting industrial welder without needing to increase welder sizes

Seems like thats the majority opinion, so as much as I hate the way they clip outside of their gridinv thats what ill do.

Nox38 commented:
@ArtisticRoomba Unfortunately, as of yet I can't find a way to stop people from refueling the emergency welder. I changed the sizes and reverted the emergency welder speed, instead just bumping up its fuel slightly.

iaada commented:
These lines on `Welder` are what give the refill behavior:
```
  - type: RefillableSolution
    solution: Welder
```
The clean solution would be to make an abstract entity with everything from `Welder` on it except this component. Then `Welder` inherits that abstract and adds `RefillableSolution`, while `WelderMini` inherits it but doesn't add the component.

Nox38 commented:
> These lines on `Welder` are what give the refill behavior:
> 
> ```
>   - type: RefillableSolution
>     solution: Welder
> ```
> 
> The clean solution would be to make an abstract entity with everything from `Welder` on it except this component. Then `Welder` inherits that abstract and adds `RefillableSolution`, while `WelderMini` inherits it but doesn't add the component.

Interestingly, when I redid the emergency welder code without a parent to welder or the refillableSolution component, it could still be refueled at a fuel tank. 

Here was the code:
- type: entity
  name: emergency welding tool
  parent: BaseItem
  id: WelderMini
  description: "A miniature, low temperature welder. Its safe to use without eye protection, but works slowly and cannot be refueled."
  components:
  - type: EmitSoundOnLand
    sound:
      path: /Audio/Items/welder_drop.ogg
  - type: Sprite
    sprite: Objects/Tools/welder_mini.rsi
    layers:
    - state: icon
    - state: welder_flame
      visible: false
      shader: unshaded
      map: [ "enum.ToggleableVisuals.Layer" ]
  - type: GenericVisualizer
    visuals:
      enum.ToggleableVisuals.Enabled:
        enum.ToggleableVisuals.Layer:
          True: { visible: true }
          False: { visible: false }
  - type: Item
    size: Tiny
    sprite: Objects/Tools/welder_mini.rsi
  - type: ItemToggle
    predictable: false
    soundActivate:
      collection: WelderOn
      params:
        variation: 0.125
        volume: -5
    soundDeactivate:
      collection: WelderOff
      params:
        variation: 0.125
        volume: -5
  - type: ItemToggleMeleeWeapon
    activatedSoundOnHit:
      path: /Audio/Weapons/Guns/Hits/energy_meat1.ogg
      params:
        variation: 0.250
        volume: -10
    activatedSoundOnHitNoDamage:
      path: /Audio/Weapons/Guns/Hits/energy_meat1.ogg
      params:
        variation: 0.250
        volume: -12
    deactivatedSoundOnHitNoDamage:
      collection: MetalThud
    activatedDamage:
      types:
        Heat: 8
  - type: ItemToggleSize
    activatedSize: Ginormous
  - type: ItemToggleHot
  - type: ComponentToggler
    components:
    - type: DisarmMalus
      malus: 0.6
  - type: ToggleableVisuals
    spriteLayer: flame
    inhandVisuals:
      left:
      - state: inhand-left-flame
        shader: unshaded
      right:
      - state: inhand-right-flame
        shader: unshaded
  - type: UseDelay
  - type: MeleeWeapon
    wideAnimationRotation: -90
    damage:
      types:
        Blunt: 2
    soundHit:
      collection: MetalThud
  - type: SolutionContainerManager
    solutions:
      Welder:
        reagents:
        - ReagentId: WeldingFuel
          Quantity: 65 # Just enough to weld two walls in a row.
        maxVol: 65
  - type: Tool
    useSound:
      collection: Welder
    qualities: Welding
    speedModifier: 0.5
  - type: Welder
  - type: PointLight
    enabled: false
    radius: 1.0
    color: orange
    netsync: false
  - type: ItemTogglePointLight
  - type: Appearance
  - type: PhysicalComposition
    materialComposition:
      Steel: 200
  - type: StaticPrice
    price: 10
  - type: IgnitionSource
    temperature: 700

iaada commented:
Ach, you're right. This is part of the weird coupling between `WelderComponent` and `ReagentTankComponent` here.
https://github.com/space-wizards/space-station-14/blob/00ce19dfb193b488ef4fb5d0b5808d4653e086a4/Content.Shared/Tools/Systems/SharedToolSystem.Welder.cs#L100

slarticodefast commented:
I brought this up for a very quick maintainer discussion and we decided to close this as low priority due to being a microbalance PR, see our list of freezes and restrictions https://github.com/space-wizards/space-station-14/issues/8524

Additionally we really need proper guidelines for roundstart equipment so that we can do a single larger balance pass instead of requiring lots of small PRs like this one, see https://github.com/space-wizards/docs/pull/524

Thank you for your contribution.
